genie-core/src/test/groovy/com/netflix/genie/core/util/ExponentialBackOffTriggerSpec.groovy

```diff
@@ -0,0 +1,147 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.core.util
+
+import com.netflix.genie.test.categories.UnitTest
+import org.junit.experimental.categories.Category
+import org.springframework.scheduling.TriggerContext
+import spock.lang.Specification
+import spock.lang.Unroll
+
+@Category(UnitTest)
+class ExponentialBackOffTriggerSpec extends Specification {
+
+    @Unroll
+    def "NextExecutionTime with type #delayType"(
+            ExponentialBackOffTrigger.DelayType delayType,
+            long[] nextExpectedExecutions
+    ) {
+        setup:
+        long min = 1L
+        long max = 16L
+        float factor = 2.0f
+        TriggerContext triggerContext = Mock(TriggerContext) {
+            lastScheduledExecutionTime() >> new Date(100)
+            lastActualExecutionTime() >> new Date(200)
+            lastCompletionTime() >> new Date(300)
+        }
+        ExponentialBackOffTrigger trigger = new ExponentialBackOffTrigger(
+                delayType,
+                min,
+                max,
+                factor
+        )
+
+        for (Long expectedDelay : nextExpectedExecutions) {
+            when:
+            Date nextExecutionTime = trigger.nextExecutionTime(triggerContext)
+
+            then:
+            Date expectedExecutionTime = new Date(expectedDelay)
+            nextExecutionTime == expectedExecutionTime
+        }
+
+        trigger.reset()
+
+        for (Long expectedDelay : nextExpectedExecutions) {
+            when:
+            Date nextExecutionTime = trigger.nextExecutionTime(triggerContext)
+
+            then:
+            Date expectedExecutionTime = new Date(expectedDelay)
+            nextExecutionTime == expectedExecutionTime
+        }
+
+        where:
+        delayType                                                              | nextExpectedExecutions
+        ExponentialBackOffTrigger.DelayType.FROM_PREVIOUS_SCHEDULING           | [101L, 102, 104, 108, 116, 116, 116]
```

Comment:
Since you left the typing in did you mean to leave this first element as a long type coercion?